rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -81,150 +90,257 @@ typedef struct
-    BoundingBoxLabels& get_bb_labels() { return _bb_label_ids; }
-    void set_bb_labels(BoundingBoxLabels bb_label_ids) {_bb_label_ids = std::move(bb_label_ids); }
-    ImgSize& get_img_size() { return _img_size; }
-    const JointsData& get_joints_data(){ return _joints_data; }
-protected:
-    BoundingBoxCords _bb_cords = {}; // For bb use
-    BoundingBoxCords_xcycwh _bb_cords_xcycwh = {}; // For bb use
-    BoundingBoxLabels _bb_label_ids = {};// For bb use
-    ImgSize _img_size = {};
-    JointsData _joints_data = {};
-    int _label_id = -1; // For label use only
+    virtual std::vector<int>& get_labels() { };
+    virtual void set_labels(Labels label_ids) { };
+    virtual BoundingBoxCords& get_bb_cords() { };
+    virtual BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { };
+    virtual void set_bb_cords_xcycwh(BoundingBoxCords_xcycwh bb_cords_xcycwh) { };
+    virtual void set_bb_cords(BoundingBoxCords bb_cords) { };
+    virtual std::vector<int>& get_polygon_count() { };
+    virtual std::vector<std::vector<int>>& get_vertices_count() { };
+    virtual MaskCords& get_mask_cords() { };
+    virtual void set_mask_cords(MaskCords mask_cords) { };
+    virtual void set_polygon_counts(std::vector<int> polygon_count) { };
+    virtual void set_vertices_counts(std::vector<std::vector<int>> vertices_count) { };
+    virtual JointsData& get_joints_data(){  };
+    ImgSize& get_img_size() {return _info.img_size; }
+    std::string& get_image_name() { return _info.img_name; }
+    uint& get_image_id() { return _info.img_id; }
+    void set_img_size(ImgSize img_size) { _info.img_size = std::move(img_size); }
+    void set_img_id(uint img_id) { _info.img_id = img_id; }
+    void set_img_name(std::string img_name) { _info.img_name = img_name; }
+    void set_metadata_info(MetaDataInfo info) { _info = std::move(info); }
+    protected:
+    MetaDataInfo _info;
 };
 
 struct Label : public MetaData
 {
-    Label(int label) { _label_id = label; }
-    Label(){ _label_id = -1; }
+    Label(int label)
+    {
+        _label_ids = {label};
+    }
+    Label()
+    {
+        _label_ids = {-1};
+    }
+    std::vector<int>& get_labels() { return _label_ids; }
+    void set_labels(Labels label_ids)
+    {
+        _label_ids = std::move(label_ids);
+    }
+    protected:
+    Labels _label_ids = {}; // For label use only
 };
 
-struct BoundingBox : public MetaData
+struct BoundingBox : public Label
 {
     BoundingBox()= default;
-    BoundingBox(BoundingBoxCords bb_cords, BoundingBoxLabels bb_label_ids)
+    BoundingBox(BoundingBoxCords bb_cords, Labels bb_label_ids, ImgSize img_size = ImgSize{0,0}, uint img_id = 0)
     {
         _bb_cords =std::move(bb_cords);
-        _bb_label_ids = std::move(bb_label_ids);
+        _label_ids = std::move(bb_label_ids);
+        _info.img_size = std::move(img_size);
+        _info.img_id = img_id;
     }
-    BoundingBox(BoundingBoxCords bb_cords, BoundingBoxLabels bb_label_ids, ImgSize img_size)
```

Comment:
What happens if support needs to be added for a dataset where there's no concept of image ID and we may need to query boxes based on file_names? Can we have a case for that?